include/bluetooth/iso.h

```diff
@@ -60,45 +60,43 @@ struct bt_iso_chan {
 	struct bt_iso_chan_ops		*ops;
 	/** Channel QoS reference */
 	struct bt_iso_chan_qos		*qos;
-	/** Channel data path reference*/
-	struct bt_iso_chan_path		*path;
 	sys_snode_t			node;
 	uint8_t				state;
 	bt_security_t			required_sec_level;
 };
 
-/** @brief Audio QoS direction */
-enum {
-	BT_ISO_CHAN_QOS_IN,
-	BT_ISO_CHAN_QOS_OUT,
-	BT_ISO_CHAN_QOS_INOUT
+/** @brief ISO Channel IO QoS structure. */
+struct bt_iso_chan_io_qos {
+	/** Channel interval in us. Value range 0x0000FF - 0x0FFFFFF. */
+	uint32_t			interval;
+	/** Channel Latency in ms. Value range 0x0005 - 0x0FA0. */
+	uint16_t			latency;
+	/** Channel SDU. Value range 0x0000 0 0x0FFF. */
+	uint8_t				sdu;
+	/** Channel PHY - See BT_GAP_LE_PHY for values.
+	 *  Setting BT_GAP_LE_PHY_NONE disables data path.
+	 */
+	uint8_t				phy;
+	/** Channel Retransmission Number. Value range 0x00 - 0x0F. */
+	uint8_t				rtn;
+	/** Channel data path reference.
+	 *  Setting to NULL default to HCI data path.
+	 */
+	struct bt_iso_chan_path		*path;
```

Comment:
The split of the QOS structure make it easier to set difference RX and TX values, which is nice. 

For the peripheral, which only sets the direction/path, this is still kind of confusing in my opinion.

Reply:
We could perhaps make the struct bt_iso_chan_path as part of bt_iso_server so the stack takes care of assigning it when accepting incoming connections.

Reply:
So the server would supply the paths by the server struct, and the client would supply the paths when setting the QOS. I think that would work fine (I don't see any immediate issues with it), if we can make the documentation very clear that the QOS is just for the initiater/central/client.

Reply:
Let do this is a separate PR though, there is already why too many conflicts everytime I check this there is a new one.